js/views/modals/Settings/Page.js

```diff
@@ -37,27 +92,46 @@ export default class extends baseVw {
     this.profile.set(formData);
 
     const save = this.profile.save();
+    let saveAvatar;
+    let saveHeader;
 
     this.trigger('saving');
 
     if (!save) {
       // client side validation failed
       this.trigger('saveComplete', true);
+      this.render();
     } else {
       this.trigger('savingToServer');
-
-      save.done(() => this.trigger('saveComplete'))
-        .fail((...args) =>
-          this.trigger('saveComplete', false, true,
-            args[0] && args[0].responseJSON && args[0].responseJSON.reason || ''));
     }
 
-    // render so errrors are shown / cleared
-    this.render();
+    if (this.avatarChanged && this.avatarCropper.cropit('imageSrc')) {
+      saveAvatar = this.saveAvatar();
+      saveAvatar.done((avatarData) => {
+        // set hash in profile to mirror the server copy
+        this.profile.set('avatarHash', avatarData.hash);
+        app.profile.set('avatarHash', avatarData.hash);
+      });
+    }
 
-    const $firstErr = this.$('.errorList:first');
+    if (this.headerChanged && this.headerCropper.cropit('imageSrc')) {
+      saveHeader = this.saveHeader();
+      saveHeader.done((headerData) => {
+        // set hash in profile to mirror the server copy
+        this.profile.set('headerHash', headerData.hash);
+        app.profile.set('headerHash', headerData.hash);
+      });
+    }
 
-    if ($firstErr.length) $firstErr[0].scrollIntoViewIfNeeded();
+    $.whenAll(save, saveAvatar, saveHeader)
+      .done(() => {
+        this.trigger('saveComplete');
+        this.render();
```

Comment:
If the user uploads an avatar or header, the view can't render until the hashes for the new images are returned, otherwise the images are set to their previous values, which is incorrect.

So if we need to render to show errors, that has to happen after the hashes are returned. 

Unless there's another way around that, we could put an overlay to block the form while the API requests are pending, like we do in the address section.